lib/maths/analytics/CBoostedTreeImpl.cc

```diff
@@ -2523,6 +2523,9 @@ core::CPackedBitVector CBoostedTreeImpl::dataSummarization(const core::CDataFram
 
     core::CPackedBitVector allTrainingRowsMask{this->allTrainingRowsMask()};
 
+    if (m_DataSummarizationFraction == 0.0) {
+        return ~allTrainingRowsMask;
```

Comment:
I think this shouldn't negate all training row mask because this will be true for test row masks. We probably get away with this because we never pass such rows now inference for them happens in Java, but I think this should be
```suggestion
        return core::CPackedBitVector{frame.numberRows(), false);
```
Also I would move to before `core::CPackedBitVector allTrainingRowsMask{this->allTrainingRowsMask()};`.